protostar/starknet_gateway/starknet_interaction.py

```diff
@@ -0,0 +1,64 @@
+from dataclasses import dataclass
+from typing import Dict, List, Optional, Union
+
+from typing_extensions import Literal
+
+from protostar.utils.log_color_provider import LogColorProvider
+
+
+@dataclass
+class StarknetInteraction:
+    direction: Literal["TO_STARKNET", "FROM_STARKNET"]
```

Comment:
I don't know if this bidirectional notation is a good one. Starknet cannot make requests to protostar. Wouldn't be better to just make requests and include responses (likein for example browser)?

Reply:
@mkaput WDYT?

Reply:
This class could be better. I designed this class mainly for the CLI logging.

```python
class StarknetInteraction:
  action: ...
  request: Payload
  reponse: Payload
```

Reply:
I mean even from user interface perspective. starknet can only respond to user requests and existing syntax suggests something that can be interpreted in a different way.

Reply:
```python
class StarknetRequest:
action: ...
payload: ...
response: ...
```

Reply:
IMO it would be better